### ISSUE TITLE
Problems unmarshaling JSON for JIRA v6.4

### ISSUE THREAD

**Issue by avadhutp** [andygrunwald/go-jira#3]
Here's my sample code:
```go
client, _ := jira.NewClient(nil, "http://my.jira.url")

if i, _, err := client.Issue.Get("DAR-1"); err != nil {
	logger.Error(err)
} else {
	logger.Info(i.Fields.Project.ID)
}
```
The errors I see are:
```
json: cannot unmarshal object into Go value of type []*jira.Worklog
json: cannot unmarshal string into Go value of type int (This is probably for issue.Fields.Updated)
json: cannot unmarshal object into Go value of type []*jira.Comment
```
The JIRA version I am using is `v6.4.10#64025-sha1:5b8b740`

Using JIRA `http://hostname/rest/api/2/issue/DAR-1`, I see that the worklog JSON looks like this:
```js
worklog: {
  startAt: 0,
  maxResults: 20,
  total: 0,
  worklogs: [ ]
},
```

**Comment by avadhutp** [comment]
Here's a more comprehensive JS from some other issue:
{
	startAt: 0,
	maxResults: 20,
	total: 1,
	worklogs: [
		{
			self: "https://jira.hostname/rest/api/2/issue/1026988/worklog/15500",
			author: {
				self: "https://jira.hostname/rest/api/2/user?username=avadhutp",
				name: "avadhutp",
				key: "avadhutp",
				emailAddress: "avadhutp@yelp.com",
			avatarUrls: {
				48x48: "https://jira.hostname/secure/useravatar?ownerId=avadhutp&avatarId=13625",
				24x24: "https://jira.hostname/secure/useravatar?size=small&ownerId=avadhutp&avatarId=13625",
				16x16: "https://jira.hostname/secure/useravatar?size=xsmall&ownerId=avadhutp&avatarId=13625",
				32x32: "https://jira.hostname/secure/useravatar?size=medium&ownerId=avadhutp&avatarId=13625"
			},
				displayName: "Avadhut Phatarpekar",
				active: true,
				timeZone: "Europe/London"
			},
			updateAuthor: {
				self: "https://jira.hostname/rest/api/2/user?username=avadhutp",
				name: "avadhutp",
				key: "avadhutp",
				emailAddress: "avadhutp@yelp.com",
			avatarUrls: {
				48x48: "https://jira.hostname/secure/useravatar?ownerId=avadhutp&avatarId=13625",
				24x24: "https://jira.hostname/secure/useravatar?size=small&ownerId=avadhutp&avatarId=13625",
				16x16: "https://jira.hostname/secure/useravatar?size=xsmall&ownerId=avadhutp&avatarId=13625",
				32x32: "https://jira.hostname/secure/useravatar?size=medium&ownerId=avadhutp&avatarId=13625"
			},
				displayName: "Avadhut Phatarpekar",
				active: true,
				timeZone: "Europe/London"
			},
			comment: "test",
			created: "2016-04-06T02:36:53.594-0700",
			updated: "2016-04-06T02:36:53.594-0700",
			started: "2016-04-06T02:36:00.000-0700",
			timeSpent: "3h",
			timeSpentSeconds: 10800,
			id: "15500"
		}
	]
}

**Comment by andygrunwald** [comment]
This was fixed in #4 
Thank you a lot!
